gen3/lib/fence/config-helper.py

```diff
@@ -0,0 +1,320 @@
+            configs = yaml.safe_load(yaml_file)
+            yaml_file.close()
+        else:
+            print((
+                "Cannot load config vars from a file with extention: {}".format(
+                    file_ext
+                )
+            ))
+    except Exception as exc:
+        # if there's any issue reading the file, exit
+        print((
+            "Error reading {}. Cannot get configuration. Skipping this file. "
+            "Details: {}".format(file_list, str(exc))
+        ))
+        raise
+
+    return configs
+
+
+def _get_all_configs(file_list):
+    """
+    Attempt to parse given list of files and extract configuration variables and values
+    """
+    additional_configs = dict()
+    for file_path in file_list:
+        configs = _load_config_file(file_path)
+        if configs:
+            additional_configs.update(configs)
+
+    return additional_configs
+
+
+def _nested_modify_config(action, config_str, key, value, traverse_path=None):
+    traverse_path = traverse_path or key
+    try:
+        for inner_key, inner_value in value.items():
+            temp_path = traverse_path
+            temp_path = temp_path + "/" + inner_key
+            config_str = _nested_modify_config(
+                action, config_str, inner_key, inner_value, temp_path
+            )
+    except AttributeError:
+        # not a dict so replace
+        if value is not None:
+            config_str = _modify_config(action, config_str, traverse_path, value)
+            
+    return config_str
+
+
+def _find_next_sibling_or_parent_position(yaml_config_str, start, nested_level):
+    """
+    Find the position of the next non-child configuration in a YAML file string.
+    Assume comments are always above the configurations.
+
+    Args:
+        yaml_config_str (str): a string representing a full configuration file
+        start (integer): start position fo the key
+        nested_level (integer): nested level count
+    """
+    lines = yaml_config_str[start:].split("\n")
+    last_end_position = start
+    last_child_end_position = start
+    for line in lines:
+        line_without_leading_spaces = line.lstrip(' ')
+        is_comment = (len(line_without_leading_spaces) > 0 and line_without_leading_spaces[0] == "#")
+        leading_spaces_cnt = len(line) - len(line.lstrip(' '))
+        is_child = (leading_spaces_cnt > nested_level * 2)
+        is_empty_line = (len(line_without_leading_spaces) == 0)
+        if not (is_empty_line or is_comment or is_child):
+            break
+        else:
+            last_end_position += len(line) + 1 # plus one because "\n"
+            if is_child:
+                last_child_end_position = last_end_position
+    return last_child_end_position
+
+
+def _find_comments_start_position(yaml_config_str, pos):
+    """
+    Find the start position of the comment block in a YAML file string.
+    Assume comments are always above the configurations.
```

Comment:
this comment would be removed

Reply:
I will add a notice to the headers of fence default config file like: "avoid using inline comments because those might be wiped by the cloud automation config helper script). And seems there's only one inline comment in fence default config (S3_BUCKETS/bucket2/region), how about I also move that comment.